### PR TITLE
Update polyfill support doc

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
 
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?
Polyfill support info out of date for v8

Issue Number: https://github.com/angular/angular/issues/31139

## What is the new behavior?
removed refs to `reflect` polyfill, changes ES6 to ES2015, general edit/organization improvements to text.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/browser-support.md [hunk 1]
```diff
@@ -189,26 +182,13 @@ These are the polyfills required to run an Angular application on each supported
...
-      [ES7/reflect](guide/browser-support#core-es7-reflect) (JIT only)
-
-    </td>
-
-  </tr>
-
-  <tr style="vertical-align: top">
-
-    <td>
+      Chrome, Firefox, Edge, Safari 9+ <br>
```
Comment (trotyl) [inline aio/content/guide/browser-support.md]: The same doc says only Safari 11+ is supported (latest 2), either it should be changed, or non-supported browsers should not be mentioned in polyfilling.

File: aio/content/guide/browser-support.md [hunk 2]
```diff
@@ -222,7 +202,7 @@ These are the polyfills required to run an Angular application on each supported
 
     <td>
 
-      [ES6<br>classList](guide/browser-support#classlist)
+      [ES2015<br>classList](guide/browser-support#classlist)
```
Comment (trotyl) [inline aio/content/guide/browser-support.md]: The link should be on `classList` only, not both.

File: aio/content/guide/browser-support.md (hunk 1 above)
Comment (jbogarthyde) [reply aio/content/guide/browser-support.md]: Simplifying version support statements. @igorMinar @alexeagle please check

**Comment by brandonroberts** [comment]
LGTM. I'll leave final approval to @IgorMinar

**Review by brandonroberts — COMMENTED** [review]
LGTM

**Comment by jbogarthyde** [comment]
Please help get codefresh passed

**Review by IgorMinar — APPROVED** [review]
lgtm, thanks!

**Pull request closed without merging**